docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,448 @@
+
+#### Managing dependencies
+
+Managing dependencies between HelmChartProxies is out of scope for this project. However, Helm charts define their dependencies in the [Chart.yaml file](https://helm.sh/docs/topics/charts/#the-chartyaml-file) and will install them automatically. If a Helm chart has a critical dependency on another chart, it is the chart owner's responsibility to enable it by default.
+
+### Implementation details 
+
+The implementation will include two CRDs and two controllers as follows. As this is a prototype, the specific details regarding the CRD specs are subject to change.
+
+#### HelmChartProxy CRD
+
+The first CRD, HelmChartProxy, serves as the user interface for defining and configuring Helm charts as well as selecting which workload clusters it will be installed on. 
+
+```go
+// HelmChartProxySpec defines the desired state of HelmChartProxy.
+type HelmChartProxySpec struct {
+	// ClusterSelector selects Clusters with a label that matches the specified key/value pair. The Helm chart will be
+	// installed on all selected Clusters. If a Cluster is no longer selected, the Helm release will be uninstalled.
+	ClusterSelector ClusterSelectorLabel `json:"clusterSelector"`
+
+	// ChartName is the name of the Helm chart in the repository.
+	ChartName string `json:"chartName,omitempty"`
+
+	// RepoURL is the URL of the Helm chart repository.
+	RepoURL string `json:"repoURL,omitempty"`
+
+	// ReleaseName is the release name of the installed Helm chart. If it is not specified, a name will be generated.
+	// +optional
+	ReleaseName string `json:"releaseName,omitempty"`
+
+	// Version is the version of the Helm chart. If it is not specified, the latest version will be used.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// Namespace is the namespace the Helm release will be installed on each selected Cluster. If it is not specified, the default namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Values is an inline YAML representing the values for the Helm chart. This YAML supports Go templating to reference
+	// fields from each selected workload Cluster and programatically create and set values.
+	// +optional
+	Values string `json:"values,omitempty"`
+}
+
+// ClusterSelectorLabel defines a key/value pair used to select Clusters with a label matching the specified key and value.
+type ClusterSelectorLabel struct {
+	// Key is the label key.
+	Key string `json:"key"`
+
+	// Value is the label value.
+	Value string `json:"value"`
+}
+
+// HelmChartProxyStatus defines the observed state of HelmChartProxy.
+type HelmChartProxyStatus struct {
+	// Ready is true when the HelmReleaseProxySpec for each selected Cluster is up to date.
+	// +optional
+	Ready bool `json:"ready"`
+
+	// MatchingClusters is the list of references to Clusters selected by the ClusterSelectorLabel.
+	// +optional
+	MatchingClusters []corev1.ObjectReference `json:"matchingClusters"`
```

Comment:
may be just .Clusters?
what's the value of this field?

Reply:
The idea is that you can verify that you didn't make a mistake by installing on a Cluster you didn't intend to or forgot to include a Cluster. Since we're selecting using labels to select, it could be easy to forget to remove a label in a Cluster for example.